project/src/main/java/com/google/sps/servlets/QueryServlet.java

```diff
@@ -48,17 +53,38 @@ void init(PlacesFetcher inputFetcher) {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ImmutableList<Place> filteredPlaces;
     try {
-      //TODO(M1): add call to filterer
-      response.setContentType("application/json");
-      response.getWriter().write(new Gson().toJson(
-          Places.randomSort(fetcher.fetch())
-          .stream()
-          .limit(MAX_NUM_PLACES_TO_RECOMMEND)
-          .collect(Collectors.toList())
-      ));
+      UserPreferences userPrefs =
+          UserPreferences.builder()
+              .setMinRating(Float.parseFloat(request.getParameter("rating")))
+              .setMaxPriceLevel(Integer.parseInt(request.getParameter("price")))
+              .setOpenNow(Integer.parseInt(request.getParameter("open")) != 0)
```

Comment:
Can this parameter be null as well?

Reply:
I'm not sure I understand your question, could you please explain? 
always 0 or 1 are sent to the parameter, it can never receive nothing. are you asking if we can change it so that it'll be null when the user doesn't mind?

Reply:
"never" is a strong word. 
You rely on the client to always provide the right input - but this means you will fail if it doesn't! 
For instance, if someone changes the client and it no longer sends this field, it will be null! This can always and does happen, so your code should be immune and not fail ungracefully.

Reply:
Sorry, I still don't understand your question. 
All the other lines can be broken as well if we don't get them from the client.

Reply:
Actually, you're right. All of those will throw some exception... So you can leave as-is, I guess.